server-state-machine.go

```diff
@@ -122,6 +123,8 @@ func (state serverStateStart) Next(hr handshakeMessageReader) (HandshakeState, [
 	clientALPN := new(ALPNExtension)
 	clientPSKModes := new(PSKKeyExchangeModesExtension)
 	clientCookie := new(CookieExtension)
+	clientClientCertType := new(ClientCertTypeExtension)
+	clientServerCertType := new(ServerCertTypeExtension)
```

Comment:
Should be:

clientClientCertType := &ClientCertTypeExtension{HandshakeType: HandshakeTypeClientHello}
clientServerCertType := &ServerCertTypeExtension{HandshakeType: HandshakeTypeClientHello}